public/src/components/market/Market.tsx

```diff
@@ -121,7 +123,7 @@ export class Market extends React.Component<MarketProps, MarketState> {
                     <td className={"volume " + diffClass}><strong>{percentageIncrease}{" %"}</strong></td>
                     <td className="volume"><strong>{currentStock.getStocksInExchange()}</strong></td>
                     <td className="volume"><strong><input id={"input-" + currentStock.getId()} placeholder="0" className="market-input" /></strong></td>
-                    <td className="volume"><strong><button disabled={this.props.isMarketOpen ? false : true} className="ui inverted green button" onClick={(e) => { this.purchaseFromExchange(e, currentStock.getId()) }}>Buy</button></strong></td>
+                    <td className="volume"><strong><button disabled={disablePanel ? true : false} className="ui inverted green button" onClick={(e) => { this.purchaseFromExchange(e, currentStock.getId()) }}>Buy</button></strong></td>
```

Comment:
Do not change these if you are redirecting to register phone everytime a user has not set his phone number.